main.py

```diff
@@ -86,28 +86,39 @@ async def recalc_ranks() -> None:
             async with redis.pipeline() as pipe:
                 for user in users:
                     inactive_days = (start_time - user["latest_pp_awarded"]) / 60 / 60 / 24
+                    country = user["country"].lower()
+
+                    rank_key = f"ripple:{redis_board}:{mode}"
+                    country_rank_key = f"{rank_key}:{country}"
+
+                    # delete all country rankings for a user besides their current country
+                    # regardless of if they are inactive or not
+                    cursor = None
+                    while cursor != 0:
+                        cursor, keys = await pipe.scan(cursor=cursor or 0, match=f"{rank_key}:*")
+
+                        for key in keys:
+                            # ensure that we never delete the key of their own country
+                            # or it may cause a temporary ranking shift
+                            if key.decode() == country_rank_key:
+                                continue
+
+                            await pipe.delete(key)
```

Comment:
this seems like it might be pretty expensive, we're 5 loops nested here, do you have a sense of impact on runtime?

Reply:
yeah that logic is somewhat expensive but i was slightly unbothered by that due to it being a cron rather than some hotpath. that said, re-reading gave me some obvious optimisations so i've rewritten how it works (and also fixed a bug in the process)

Reply:
the more i interface with rankings, the more i think about changing how our ranks work to be similar to osu! (and also centralising this stuff, the old score processor idea i've talked about a few times)